src/controllers/communityController.js

```diff
@@ -0,0 +1,219 @@
+const community_trips_cache = [];
+const community_cache = [];
+//TODO: create a community cache
+//TODO: return results using the utils defined by Varun
+
+
+
+const client = require("../config/configDB");
+
+
+export const getAllCommunities = async (req, res) => {
+    client.query("SELECT * FROM communities", function(error, results, fields) {
+        if (!error) {
+            res.status(200).json(results);
+        } else {
+            res.status(500).json({
+                code: 500,
+                message: "unexpected error",
+            });
+        }
+    });
+};
+
+export const createCommunity = async (req, res) => {
+    client.query(
+        "INSERT INTO communities (community_name, community_desc) VALUES ($1, $2 )",
+        [
+            req.body.community_name,
+            req.body.community_desc,
+        ],
+        function(error, results, fields) {
+            if (!error) {
+                res.status(201).send(results);
+            } else {
+                console.log(error);
+                res.status(500).json({
+                    code: 500,
+                    message: "unexpected error",
+                });
+            }
+        }
+
+
+    )
+}
+export const getCommunityById = async () => {
+
+
+    client.query("SELECT *  FROM communities WHERE community_id = $1"
+        , [req.params.community_id],
+        function(error, results, fields) {
+            if (!error) {
+                res.status(200).json(results);
+            } else {
+                res.status(404).json({
+                    code: 404,
+                    message: "community not found ",
+                })
+            }
+        })
+}
+export const updateCommunity = async (req, res) => {
+    client.query(
+        "UPDATE communities SET community_name = $1, community_desc= $2 WHERE id =$3",
+        [
+            req.body.community_name,
+            req.body.community_desc,
+            req.params.community_id
+        ],
+        function(error, results, fields) {
+            if (!error) {
+                res.status(204).send(results);
+            } else if (results.length == 0) {
+                res.status(404).json({
+                    code: 404,
+                    message: "community not found",
+                });
+            } else {
+                res.status(400).json({ code: 400, message: "invalid input", })
+            }
+        }
+    );
+};
+
+export const deleteCommunity = async (req, res) => {
+    client.query(
+        "DELETE FROM communities WHERE id = $1",
+        [req.params.community_id],
+        function(error, results, fields) {
+            if (!error) {
+                res.status(204);
+            } else {
+                res.status(400).json({
+                    code: 400,
+                    message: "community not found",
+                });
+            }
+        }
+    );
+};
+
+export const getAllTripsOfCommunity = async (req, res) => {
```

Comment:
You can remove this function as I created it in my PR where I applied the functionality of applying different filters like user can check put the filter of origin, destination, date and time range it is a fully tested function which works as expected.
While this function only gives the trips and user can't query trips

Reply:
That's great! the aim of this PR was just to give comprehensive overview of the architecture. I will make the changes as necessary.